src/main/java/com/browserstack/automate/ci/jenkins/BrowserStackReportPublisher.java

```diff
@@ -0,0 +1,87 @@
+package com.browserstack.automate.ci.jenkins;
+
+import com.browserstack.automate.ci.common.BrowserStackEnvVars;
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.AbstractProject;
+import hudson.model.Job;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.BuildStepMonitor;
+import hudson.tasks.Publisher;
+import hudson.tasks.Recorder;
+import hudson.util.FormValidation;
+import jenkins.tasks.SimpleBuildStep;
+import org.kohsuke.stapler.AncestorInPath;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+import javax.annotation.Nonnull;
+import javax.servlet.ServletException;
+
+import com.browserstack.automate.ci.common.enums.ProjectType;
+
+import java.io.IOException;
+import java.io.PrintStream;
+
+public class BrowserStackReportPublisher extends Recorder implements SimpleBuildStep {
+    private PrintStream logger;
```

Comment:
You can convert this to local variable generally over here you define class variables. Looking this it gives thought that it will be initialized in constructor but that is not the case here...

Reply:
moved as local variable...